src/puppetlabs/trapperkeeper/services/authorization/authorization_core.clj

```diff
@@ -155,12 +155,24 @@
                  (str/join "', '" (sort valid-methods)) "'"))))
   (when (= "regex" (-> rule :match-request :type name str/lower-case))
     (try
-      (re-pattern (:path (:match-request rule)))
+      (let [pattern (re-pattern (:path (:match-request rule)))
+            group-count (-> pattern (.matcher "") .groupCount)
+            no-brs? (fn [matches]
+                      (not-any? string? (map #(re-find #"\$\d" %)
+                                             (flatten [(or matches "")]))))
+            has-brs (not (or (no-brs? (:allow rule))
+                             (no-brs? (:deny rule))))]
+        (if (and (not has-brs) (= group-count 0))
```

Comment:
I was hoping we could use something here that would be more comprehensive.  This would appear to help for the case where an allow or deny directive has a backreference and there are no capture groups in the regex.  But the bad error message could also come up in the case where the regex does have capture groups but a backreference refers to a non-existent capture group.

For example, I might have "$1 blah $6 blah $2" in an allow ACE with a regex of "^/puppet/v3/catalog/([^/]+)/(.*)$".  This regex has two capture groups so the $1 and $2 backreferences are valid but the $6 one is not.  I was hoping we could display the friendly, validation-time error for this case as well.

I was thinking that the algorithm would walk through each of the ACEs in the allow and deny directives and flag an error for each one found which exceeds the number of available capture groups.  For simplicity, if it were to just error out after the first one found which exceeds the number of capture groups, I'd be fine with that, too, although showing all of the offending ACEs could be a little more helpful for a user.

Does this make sense?